### PR TITLE
Add additional cleanup step to cleanup-buildenv-build-dir.py

### PR DESCRIPTION
**Pull request by hacst**
Our build environments also create lots of pdbs that might be
useful during the creation of the environment (or inconvenient
to not create) but that are not needed later on. A big example
of these are pdbs for the tests the libraries built.

As we do not want to have to distribute these with our build
environments this patch contains a list of globs that try to
match the biggest known instances of PDBs we do not need later
on.

This patch also adds a number of commandline arguments like
verbose and dry-run to be able to interact with the script
with more confidence during testing and operation.

### CODE REVIEW

File: tools/cleanup-buildenv-build-dir.py [hunk 1]
```diff
@@ -103,6 +130,10 @@ def rm(fn):
 	On all other systems, calling this function
 	is equivalent to calling os.remove().
 	'''
+	
+	if verbose: print("Removing " + fn)
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: Newline + indent?

File: tools/cleanup-buildenv-build-dir.py [hunk 2]
```diff
@@ -103,6 +130,10 @@ def rm(fn):
...
+	if dryrun: return
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: Newline + indent?

File: tools/cleanup-buildenv-build-dir.py [hunk 3]
```diff
@@ -171,15 +202,49 @@ def removeFilesInDir(targetDir):
...
+	parser.add_argument('builddir', help='<buildenv-name>.build directory (e.g. c:/MumbleBuild/centos-ermine-1.2.x-2014-06-01-cf59267.build)')
+	parser.add_argument('--dry-run', action='store_true')
+	parser.add_argument('-v', '--verbose', action='store_true')
+	parser.add_argument('--force' , action='store_true', help = 'Force run on directories not ending in .build')
+	args = parser.parse_args();
+	verbose = args.verbose
+	dryrun = args.dry_run
+	build_dir = args.builddir
+	
+	if verbose: print("Performing cleanup on " + build_dir)
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: Newline + indent?

File: tools/cleanup-buildenv-build-dir.py [hunk 4]
```diff
@@ -171,15 +202,49 @@ def removeFilesInDir(targetDir):
...
+	if dryrun: print("Performing dry run. No files will actually be deleted.")
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: Newline + indent?

File: tools/cleanup-buildenv-build-dir.py [hunk 5]
```diff
@@ -171,15 +202,49 @@ def removeFilesInDir(targetDir):
...
+	
+	if not args.force:
+		build_dirname =  os.path.basename(os.path.normpath(build_dir))
+		if not build_dirname.endswith(".build"):
+			print("Build directory name '" + build_dirname + "' does not end in .build . If you are sure you want to clean this path re-execute with --force")
+			sys.exit(1)
+	
+	# Perform initial cleanup
+	if verbose: print("Performing initial cleanup")
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: Newline + indent?

File: tools/cleanup-buildenv-build-dir.py [hunk 6]
```diff
@@ -171,15 +202,49 @@ def removeFilesInDir(targetDir):
...
 	removeFilesInDir(build_dir)
-
+	if verbose: print("Done with initial cleanup")
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: Newline + indent?

File: tools/cleanup-buildenv-build-dir.py [hunk 7]
```diff
@@ -171,15 +202,49 @@ def removeFilesInDir(targetDir):
...
+	
+	# After that perform additional cleanup
+	if verbose: print("Performing additional cleanup")
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: Newline + indent?

File: tools/cleanup-buildenv-build-dir.py [hunk 8]
```diff
@@ -171,15 +202,49 @@ def removeFilesInDir(targetDir):
...
+	# !!!!!
+	KEEP_EXT = [] # !!! We do not obey preversed extensions after this point !!!
```
Comment (mkrautz) [inline tools/cleanup-buildenv-build-dir.py]: How about we add a keep_ext parameter to removeFilesInDir to avoid the global, and this scary block?

Comment (hacst) [inline tools/cleanup-buildenv-build-dir.py]: Fine ;) Thought about it and then though "well...how much functionality are we actually going to add to this one...". Also verbose and dryrun are also globals. KEEP_EXT surely is the ugliest by far though and easy enough to fix.

**Pull request merged**